sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/Logger.kt

```diff
@@ -0,0 +1,220 @@
+                }
+            }
+        }
+    }
+
+    private fun invocationLocation(): String {
+        // Appends the location of the call to #debug, #info, etc…, i.e. 2 levels down the current.
+        val st = Thread.currentThread().stackTrace
+        var location: StackTraceElement? = null
+        for (i in min(2, st.size) until st.size) {
+            if (st[i].className != this.javaClass.name) {
+                location = st[i]
+                break
+            }
+        }
+        return if (location != null) "${location.className}#${location.methodName}:${location.lineNumber} " else ""
+    }
+
+    /**
+     * Log messages as DEBUG.
+     * Will use [String.format] to format the message by substituting the given arguments [args].
+     * @param message: The message to log.
+     * @param args: Arguments for the format string.
+     */
+    fun debug(message: String?, vararg args: Any?) {
+        log(Log.DEBUG, null, message, *args)
+    }
+
+    /**
+     * Log messages as DEBUG.
+     * Will use [String.format] to format the message by substituting the given arguments [args].
+     * @param message: The message to log.
+     * @param cause: The given exception to log.
+     * @param args: Arguments for the format string.
+     */
+    fun debug(message: String?, cause: Throwable?, vararg args: Any?) {
+        log(Log.DEBUG, cause, message, *args)
+    }
+
+    /**
+     * Log messages as DEBUG.
+     * Will use [String.format] to format the message by substituting the given arguments [args].
+     * @param cause: The given exception to log.
+     * @param message: The message to log.
+     * @param args: Arguments for the format string.
+     */
+    fun debug(cause: Throwable? = null, message: String?, vararg args: Any?) {
+        log(Log.DEBUG, cause, message, *args)
+    }
+
+    /**
+     * Log messages as INFO.
+     * * Will use [String.format] to format the message by substituting the given arguments [args].
+     * @param message: The message to log.
+     * @param args: Arguments for the format string.
+     */
+    fun info(message: String?, vararg args: Any?) {
+        log(Log.INFO, null, message, *args)
+    }
+
+    /**
+     * Log messages as INFO.
+     * Will use [String.format] to format the message by substituting the arguments.
+     * @param cause: The given exception to log.
+     * @param message: The message to log.
+     * @param args: Arguments for the format string.
+     */
+    fun info(message: String?, cause: Throwable?, vararg args: Any?) {
+        log(Log.INFO, cause, message, *args)
+    }
+
+    /**
+     * Log messages as INFO.
+     * Will use [String.format] to format the message by substituting the arguments.
+     * @param cause: The given exception to log.
+     * @param message: The message to log.
+     * @param args: Arguments for the format string.
+     */
+    fun info(cause: Throwable? = null, message: String?, vararg args: Any?) {
```

Comment:
Do we need to keep both 'info' method with same 3 params with different sequence?

Reply:
Deleted the duplicated methods